From 6120c556f8de5cc837832fe0d46b4fd4f88618ba Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Fri, 20 Oct 2023 04:54:18 +0800
Subject: [PATCH] copyfile_stubs.c: define COPYFILE_CLONE in case it is
 undefined

Credit to Ali Caglayan <alizter@gmail.com>
Picked from https://github.com/ocaml/dune/pull/8942/commits/89671ada56e24af7654a5b57b9089501342a2e62
---
 otherlibs/stdune/src/copyfile_stubs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git otherlibs/stdune/src/copyfile_stubs.c otherlibs/stdune/src/copyfile_stubs.c
index 09c9a1b14..6824c572f 100644
--- otherlibs/stdune/src/copyfile_stubs.c
+++ otherlibs/stdune/src/copyfile_stubs.c
@@ -15,6 +15,10 @@
 #include <string.h>
 #include <sys/syslimits.h>
 
+#ifndef COPYFILE_CLONE
+#define COPYFILE_CLONE (1<<24)
+#endif
+
 CAMLprim value stdune_copyfile(value v_from, value v_to) {
   CAMLparam2(v_from, v_to);
   caml_unix_check_path(v_from, "copyfile");
